src/ext/handlersToResp.c

```diff
@@ -89,9 +125,17 @@ static void deleteRdbToRespCtx(RdbParser *p, void *context) {
     /* ignore the first release attempt */
     if (--ctx->refcount) return;
 
+    if(ctx->streamCtx.grpName)
+        RDB_bulkCopyFree(p, ctx->streamCtx.grpName);
+
+    if(ctx->streamCtx.groupPel)
+        deletePendingEntriesList(p, &ctx->streamCtx.groupPel);
+
     if (ctx->keyCtx.key != NULL)
         RDB_bulkCopyFree(p, ctx->keyCtx.key);
-    ctx->keyCtx.key = NULL;
+
+    if(ctx->streamCtx.consName)
+        RDB_bulkCopyFree(p, ctx->streamCtx.consName);
```

Comment:
Maybe you can consider adding null check into `RDB_bulkCopyFree()`. So, you won't have to check it on each call and it would be in line with stdlib `free()` as we can call it null ptr. We can also provide same behavior for other `free()` like functions in the API.

Reply:
Right. thanks.